### PR TITLE
Fix #7

### LINKED ISSUE #7: Feedback on the fine-tuning notebook
@vasudevgupta7 great work thus far. Here ate some more pointers. 

"How to train TensorFlow saved-model with extra head", I suggest "Fine-tuning <model name> with an extra head". 

"In this notebook, we will load the pre-trained wav2vec2 model from TFHub and will train it on LibriSpeech dataset by appending LM head over the top of our pre-trained model.", I suggest something like - 

> In this notebook, we will load the pre-trained wav2vec2 model from TFHub (should be a link to the model when available) and will train it on LibriSpeech dataset by appending LM head over the top of our pre-trained model. The underlying task is to ...

"You can also refer to this repositary for some more amazing tutorials on speech-related tasks. In case you encountered any bug in this notebook, please create an issue here."

Typo.

**Additional feedback**: 
* Let's try to wrap the training and evaluation loop as a subclassed model (`tf.keras.Model`). Let us know if you face any problems there. 
* Is it possible to load a few FLAC files in an Audio widget and play them for reference? See if this [Colab Notebook](https://colab.research.google.com/github/sayakpaul/GCP-ML-API-Demos/blob/master/Abstract_Parser.ipynb) helps. 

Cc: @MorganR

### PR DESCRIPTION
**Pull request by vasudevgupta7**
### Major changes:

* Added section for wrapping training with `tf.keras.Model`
* added audio widget
* some text is also fixed

Link to notebook: https://colab.research.google.com/github/vasudevgupta7/gsoc-wav2vec2/blob/export-v3/notebooks/wav2vec2_saved_model_finetuning.ipynb

@sayakpaul @MorganR

### CODE REVIEW

**Comment by sayakpaul** [comment]
@vasudevgupta7 this looks great:

image

Instead of prefixing this sample, could we choose that at random? It'd introduce a little bit of variety. 

About the following note:

> Note: For distributed training (especially on TPUs), .from_generator(...) doesn't work currently & it is recommended to train on data stored in .tfrecord format.

The TFRecords should ideally be _stored inside a GCS Bucket_ in order for the TPUs to work to the fullest extent. 

> metirc.compute() will calculate the final WER score over all the batches added in previous cell.

Typo.

> This notebook is just for showing the workflow of training a saved model.

We are fine-tuning the model, no? How about changing it to something like - "This notebook gives you a template to fine-tune a pre-trained speech model."

Additional suggestions:

* Let's try to have a separate validation split as well to ensure the model is learning. 
* Let's try to add the metric to `compile()` to make it even more end to end. 
* Let's add an inference code snippet as well. 
* `KerasLayer` makes the model a bit opaque which can be a concern among practitioners in case they want finer controls. If the `SavedModel` is loaded with `tf.keras.models.load_model()` it actually gets expanded as a full-blown model providing finer controls as expected. Do you think we should add a note on this? 

Overall, I think this is GREAT WORK! After the model is published from Hub I'd suggest working with the Hub folks to add this Colab here: https://github.com/tensorflow/hub/tree/master/examples.

**Comment by vasudevgupta7** [comment]
@sayakpaul I am not sure how can we add metric to `.compile` as `WER` is not available in TensorFlow & I was using HuggingFace WER which can't be used inside `test_step`.

I have completed rest of the suggestions.

**Comment by sayakpaul** [comment]
@vasudevgupta7 if it's not too much trouble is it possible to implement WER and pass that as a metric to the `metrics` argument of `compile()`? 

You can refer to [this notebook](https://www.kaggle.com/spsayakpaul/funmatch-distillation/) as an example to see how I am passing a stock metric to `compile()` and how it is getting used to evaluate the model being trained.

**Comment by vasudevgupta7** [comment]
Okay, I will try to implement WER.

Merging this PR for now as we can have further discussion on `Hub` repo: https://github.com/tensorflow/hub/pull/788
